compiler/ilgen/MethodBuilder.hpp

```diff
@@ -73,7 +74,7 @@ class MethodBuilder : public TR::IlBuilder
 
    TR::TypeDictionary *typeDictionary()                      { return _types; }
 
-   const char *getDefiningFile()                             { return _definingFile; }
+   std::string getDefiningFile()                             { return _definingFile; }
```

Comment:
Making it return by reference would certainly be more efficient. I intended to make it return by value to avoid the possibilities of accidental modification and dangling reference. However, if you prefer, I could make it return a const reference, which would help us _get around_ the first problem.

As far as allocation is concerned, the copy would get allocated by the caller.